File: router/router-annotation/src/main/java/com/sofar/router/annotation/RouterService.java
package com.sofar.router.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.CLASS)
public @interface RouterService {

  Class interfaces();

  /**
   * 获取Service实例时，是否是单例
   */
  boolean singleton() default false;
}

File: gradle-8x/router/router-annotation/src/main/java/com/sofar/router/annotation/RouterService.kt
package com.sofar.router.annotation

import kotlin.reflect.KClass

@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.BINARY) // 对应 Java 的 CLASS 级别
annotation class RouterService(
  /**
   * 指定接口类型
   */
  val interfaces: KClass<*>,

  /**
   * 获取 Service 实例时，是否是单例
   */
  val singleton: Boolean = false
)